### PR TITLE
Allow user to change music in menu

### PR DESCRIPTION
**Pull request by EmeraldLoc**
The word "Music" does need translation, since it's 1 word, should I just use google translate?

I tried to make it as simple as it can be to add/remove songs to the main menu. All you have to do is edit the `gMainMenuSounds`, everything else is done for you.

`gMainMenuSounds` is a structure of a new type, `MainMenuSounds`, which contains a name, and a sound. `configMenuSounds` was changed to a `int`, and it now references an entry in the `gMainMenuSounds` table. The stuff done in `level_update.c` is very self explanatory. Stage music is handled with a custom definition, `STAGE_MUSIC`, that's set to the integer `0`. This is also passed into `gMainMenuSounds`. Something very important is that the stage music entry needs to ALWAYS be last in the table, this will be discussed later in this message. The main part I'm not super happy with is getting staff roll integration to work. When staff roll is toggled, I chose, as I don't know of any other methods, to recreate the panel. The reason is that the selection box's selection needs to be updated, and whether or not `Stage Music` exists needs to be updated as well.

### CODE REVIEW

**Comment by Cooliokid956** [comment]
> The word "Music" does need translation, since it's 1 word, should I just use google translate?

The word "Music" is available in `[SOUND]:MUSIC_VOLUME`, use the power of context clues to find the word "Music" in some of the trickier translations!

**Comment by Flower35** [comment]
> > The word "Music" does need translation, since it's 1 word, should I just use google translate?
> 
> The word "Music" is available in `[SOUND]:MUSIC_VOLUME`, use the power of context clues to find the word "Music" in some of the trickier translations!

Sorry, that's actually a terrible advice 😃

Quoting Wikipedia:

> In linguistics, declension (verb: to decline) is the changing of the form of a word, generally to express its syntactic function in the sentence, by way of some inflection.
 
> Unlike English, many languages use suffixes to specify subjects and objects and word cases in general.

---

And an example from `Polish.ini`:
```
MUSIC_VOLUME = "Głośność Muzyki"
```
* (Genitive Case - Dopełniacz): _„Muzyki”_

Here, _„Głośność Muzyki”_ literally (and naturally) translates as _"Volume of the Music"_

---

For "Music" as a single noun, the correct translation will be:
```
MUSIC = "Muzyka"
```
* (Nominative Case - Mianownik): _„Muzyka”_

---

Similarly for `Czech.ini` (`MUSIC = "písnička"`) and `Russian.ini` (`MUSIC = "музыка"`), although it would be nice if some native speaker / translator could verify these. 🙂

File: lang/Polish.ini [hunk 1]
```diff
@@ -260,7 +260,7 @@ QUIT = "Wyjdź"
 MENU_TITLE = "MENU"
 LEVEL = "Poziom"
 STAFF_ROLL = "Napisy Końcowe"
-USE_STAGE_MUSIC = "Użyj Muzyki z Poziomu"
+MUSIC = "Muzyki"
```
Comment (Flower35) [inline lang/Polish.ini]:
```suggestion
MUSIC = "Muzyka"
```

File: lang/Russian.ini [hunk 2]
```diff
@@ -259,7 +259,7 @@ QUIT = "Выход"
 MENU_TITLE = "MENU"
 LEVEL = "Уровень"
 STAFF_ROLL = "Завершение"
-USE_STAGE_MUSIC = "Использовать музыку уровня"
+MUSIC = "музыки"
```
Comment (Flower35) [inline lang/Russian.ini]:
```suggestion
MUSIC = "Музыка"
```

File: lang/Czech.ini [hunk 3]
```diff
@@ -260,7 +260,7 @@ QUIT = "Opustit hru"
 MENU_TITLE = "MENU"
 LEVEL = "Level"
 STAFF_ROLL = "Závěrečné titulky"
-USE_STAGE_MUSIC = "Použít písničku z levelu"
+MUSIC = "Hlasitost"
```
Comment (Flower35) [inline lang/Czech.ini]:
```suggestion
MUSIC = "Písnička"
```

**Review by Flower35 — CHANGES_REQUESTED** [review]

File: lang/Polish.ini (hunk 1 above)
Comment (EmeraldLoc) [reply lang/Polish.ini]: Thank you!

**Comment by AgentXLP** [comment]
> The word "Music" is available in `[SOUND]:MUSIC_VOLUME`, use the power of context clues to find the word "Music" in some of the trickier translations!

Lol

**Comment by frog8412** [comment]
> > > The word "Music" does need translation, since it's 1 word, should I just use google translate?
> > 
> > 
> > The word "Music" is available in `[SOUND]:MUSIC_VOLUME`, use the power of context clues to find the word "Music" in some of the trickier translations!
> 
> Sorry, that's actually a terrible advice 😃
> 
> Quoting Wikipedia:
> 
> > In linguistics, declension (verb: to decline) is the changing of the form of a word, generally to express its syntactic function in the sentence, by way of some inflection.
> 
> > Unlike English, many languages use suffixes to specify subjects and objects and word cases in general.
> 
> And an example from `Polish.ini`:
> 
> ```
> MUSIC_VOLUME = "Głośność Muzyki"
> ```
> 
> * (Genitive Case - Dopełniacz): _„Muzyki”_
> 
> Here, _„Głośność Muzyki”_ literally (and naturally) translates as _"Volume of the Music"_
> 
> For "Music" as a single noun, the correct translation will be:
> 
> ```
> MUSIC = "Muzyka"
> ```
> 
> * (Nominative Case - Mianownik): _„Muzyka”_
> 
> Similarly for `Czech.ini` (`MUSIC = "písnička"`) and `Russian.ini` (`MUSIC = "музыка"`), although it would be nice if some native speaker / translator could verify these. 🙂

Yep, this is correct, I'm a native Russian speaker and that is also the same in Russian (громкость музыки - музыка)

**Pull request merged**